.github/actions/go-test/action.yml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2024 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'go-test'
+
+inputs:
+  go_version:
+    description: 'The version of Go to install and use.'
+    type: 'string'
+  go_version_file:
+    description: 'Path to the go.mod file to extract a version.'
+    type: 'string'
+    default: 'go.mod'
+  go_packages:
+    description: 'List of go packages to test.'
+    type: 'string'
+    default: './...'
+  go_proxy:
+    description: 'Proxy script to wrap go commands.'
+    type: 'string'
+    default: 'go'
+  go_test_timeout:
+    description: 'Timeout for Go tests.'
+    type: 'string'
+    default: '10m'
+  directory:
+    description: 'Directory to the root of the Go module.'
+    type: 'string'
+    default: '.'
```

Comment:
While we're here, it seems like the community has standardized on "kebab" instead of "snake" case: https://dev.to/n3wt0n/composite-actions-vs-reusable-workflows-what-is-the-difference-github-actions-11kd